File: sdkwork-backend-sdk-typescript/src/types/plus-category-form.ts
/** Category creation form */
export interface PlusCategoryForm {
  /** Category name */
  name: string;
  /** Category description */
  description?: string;
  /** Category type */
  type: 'DEFAULT' | 'AGENT' | 'TOOL' | 'PROMPT' | 'FEEDS' | 'PRODUCT' | 'COURSE' | 'NEWS' | 'BOOK' | 'VIDEO' | 'AUDIO' | 'MUSIC' | 'SAAS' | 'PPT' | 'EVENTS' | 'NOTARY' | 'SHOP' | 'VOICE_SPEAK' | 'SOUND_EFFECT' | 'SKILLS' | 'SKILLS_COLLECTION' | 'PLUGIN' | 'MCP' | 'KNOWLEDGE_BASE' | 'RECRUIT' | 'CLAW' | 'OTHER';
  /** Category icon */
  icon?: string;
  /** Sort weight */
  sortWeight?: number;
  /** Parent category ID */
  parentId?: number;
  /** Category path */
  path?: string[];
  /** Visibility flag */
  visible: number;
  /** Category status */
  status: 'DEFAULT' | 'ACTIVE' | 'INACTIVE' | 'DELETED';
}

File: sdkwork-backend-sdk-go/types/plus_category_form.go
package types

// Category creation form
type PlusCategoryForm struct {
	Name string `json:"name"`
	Description string `json:"description"`
	Type string `json:"type"`
	Icon string `json:"icon"`
	SortWeight int `json:"sortWeight"`
	ParentId int `json:"parentId"`
	Path []string `json:"path"`
	Visible int `json:"visible"`
	Status string `json:"status"`
}
